File: generated-code/example/swift/Sources/TransGenTest/Structure.swift
/// Example structure
public struct Structure {
    /// Text
    let text: String

    /// 32-bit float
    let floatNumber: Float

    /// 64-bit float
    let doubleNumber: Double

    /// Read Structure from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Structure {
        var text: String
        text = stream.readString()
        var floatNumber: Float
        floatNumber = stream.readFloat()
        var doubleNumber: Double
        doubleNumber = stream.readDouble()
        return Structure(text: text, floatNumber: floatNumber, doubleNumber: doubleNumber)
    }

    /// Write Structure to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeString(text)
        stream.writeFloat(floatNumber)
        stream.writeDouble(doubleNumber)
    }
}

File: generated-code/kotlin/example/FileReadWrite/src/main/kotlin/trans_gen_test/Structure.kt
package trans_gen_test

import trans_gen_test.util.StreamUtil

/**
 * Example structure
 */
class Structure {
    /**
     * Text
     */
    var text: String
    /**
     * 32-bit float
     */
    var floatNumber: Float
    /**
     * 64-bit float
     */
    var doubleNumber: Double

    constructor(text: String, floatNumber: Float, doubleNumber: Double) {
        this.text = text
        this.floatNumber = floatNumber
        this.doubleNumber = doubleNumber
    }

    /**
     * Write Structure to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeString(stream, text)
        StreamUtil.writeFloat(stream, floatNumber)
        StreamUtil.writeDouble(stream, doubleNumber)
    }

    /**
     * Get string representation of Structure
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Structure { ")
        stringBuilder.append("text: ")
        stringBuilder.append('"' + text + '"')
        stringBuilder.append(", ")
        stringBuilder.append("floatNumber: ")
        stringBuilder.append(floatNumber)
        stringBuilder.append(", ")
        stringBuilder.append("doubleNumber: ")
        stringBuilder.append(doubleNumber)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Structure from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Structure {
            var text: String
            text = StreamUtil.readString(stream)
            var floatNumber: Float
            floatNumber = StreamUtil.readFloat(stream)
            var doubleNumber: Double
            doubleNumber = StreamUtil.readDouble(stream)
            return Structure(text, floatNumber, doubleNumber)
        }
    }
}